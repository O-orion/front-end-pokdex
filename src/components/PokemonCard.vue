<script setup lang="ts">
import { TypesPokemon } from '../types/Types.enum';

    interface PokemonCardProps {
        name: string;
        image: string
        id: number
        types: any[]
    }

    defineProps<PokemonCardProps>()

    const getTypeColor = (typeName: string): string => {
        return TypesPokemon[typeName as keyof typeof TypesPokemon]  || '#000'
    }

    const capitalizeFirstLetter = (name: string): string => {
        return name.charAt(0).toLocaleUpperCase() + name.slice(1).toLowerCase()
    }

</script>

<template>
    <section class="card-pokemon" >
        <img  class="imagem-pokemon" :src="image" :alt="name">

        <div class="informacoes-pokemon">
            <p class="nome-pokemon" >{{ capitalizeFirstLetter(name) }}</p>
            <p class="numero-pokemon" >#000{{ id }}</p>
        </div>
        <div class="pokemon-tags">
            <p v-for="type in types" :style="{ backgroundColor: getTypeColor(type.type.name) }">
                {{ capitalizeFirstLetter(type.type.name)  }}
            </p>
        </div>
    </section>
</template>

<style src="../styles/components/pokemonCard.css" ></style>
