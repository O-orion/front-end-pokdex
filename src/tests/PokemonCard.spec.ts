import { mount } from "@vue/test-utils";
import PokemonCard from "../components/PokemonCard.vue";

describe("PokemonCard.vue", () => {
  it("deve exibir o nome do pokémon corretamente", () => {
    const wrapper = mount(PokemonCard, {
      props: {
        name: "pikachu",
        image: "https://example.com/pikachu.png",
        id: 25,
        types: [{ type: { name: "electric" } }],
      },
    });

    const pokemonName = wrapper.find(".nome-pokemon").text();
    expect(pokemonName).toBe("Pikachu");
  });

  // Testando se o ID do Pokémon é exibido
  it("deve renderizar a númeração do pokemon", () => {
    const wrapper = mount(PokemonCard, {
      props: {
        name: "bulbasaur",
        image: "https://example.com/pikachu.png",
        id: 1,
        types: [{ type: { name: "grass" } }, { type: { name: "poison" } }],
      },
    });

    const pokemonNumeracao = wrapper.find(".numero-pokemon").text();
    expect(pokemonNumeracao).toBe("#0001");
  });

  // Testando se o tipo do pokemon é renderizado com a cor correta
  it("deve renderizar o tipo do pokemon com a cor correta", () => {
    const wrapper = mount(PokemonCard, {
      props: {
        name: "charmander",
        image: "https://example.com/charmander.png",
        id: 4,
        types: [{ type: { name: "fire" } }],
      },
    });

    const typeTag = wrapper.find(".pokemon-tags p");
    const typeStyle = typeTag.attributes("style");

    expect(typeStyle).toContain("background-color: rgb(240, 128, 48)");
  });

  // Testando se os tipos estão sendo renderizados
  it("deve renderizar múltiplos tipos de Pokémon corretamente", () => {
    const wrapper = mount(PokemonCard, {
      props: {
        name: "venusaur",
        image: "https://example.com/venusaur.png",
        id: 3,
        types: [{ type: { name: "grass" } }, { type: { name: "poison" } }],
      },
    });

    const pokemonTypes = wrapper.findAll(".pokemon-tags p");

    // Verificando se possui dois tipos
    expect(pokemonTypes.length).toBe(2);

    // Primeiro tipo
    expect(pokemonTypes[0].text()).toBe("Grass");

    // Segundo tipo
    expect(pokemonTypes[1].text()).toBe("Poison");
  });

  it("deve renderizar o componente corretamente (snapshot)", () => {
    const wrapper = mount(PokemonCard, {
      props: {
        name: "squirtle",
        image: "https://example.com/squirtle.png",
        id: 7,
        types: [{ type: { name: "water" } }],
      },
    });
  
    expect(wrapper.html()).toMatchSnapshot();
  });
  

});
