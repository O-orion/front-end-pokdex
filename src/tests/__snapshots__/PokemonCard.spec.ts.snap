// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard.vue > deve renderizar o componente corretamente (snapshot) 1`] = `
"<section class="card-pokemon"><img class="imagem-pokemon" src="https://example.com/squirtle.png" alt="squirtle">
  <div class="informacoes-pokemon">
    <p class="nome-pokemon">Squirtle</p>
    <p class="numero-pokemon">#0007</p>
  </div>
  <div class="pokemon-tags">
    <p style="background-color: rgb(104, 144, 240);">Water</p>
  </div>
</section>"
`;
